### PR TITLE
feat: Pool size & running

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by ronag** [comment]
Tests needs fixing

**Comment by ronag** [comment]
@dnlup Feel free to sort out the tests here if you want. But I was referring to the failures on master.

**Comment by ronag** [comment]
@mcollina

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by ronag** [comment]
merge after https://github.com/nodejs/undici/pull/541

File: lib/pool.js [hunk 1]
```diff
@@ -59,17 +59,15 @@ class Pool extends EventEmitter {
...
 
-      if (queue.isEmpty()) {
-        if (pool[kNeedDrain]) {
-          pool[kNeedDrain] = false
-          pool.emit('drain')
-        }
+      if (pool[kNeedDrain] && !this.busy) {
+        pool[kNeedDrain] = false
+        pool.emit('drain')
+      }
```
Comment (mcollina) [inline lib/pool.js:65]: why this if was removed?

**Review by mcollina — CHANGES_REQUESTED** [review]

File: lib/pool.js (hunk 1 above)
Comment (ronag) [reply lib/pool.js:65]: I don't understand the question.

Comment (mcollina) [reply lib/pool.js:65]: you removed the `if (queue.isEmpty())` check. Why?

Comment (ronag) [reply lib/pool.js:65]: I replaced it with `this.busy` which contains the same check through `this[kPending] > 0`

Comment (ronag) [reply lib/pool.js:65]: the empty check is moved to just `if (pool[kClosedResolve] && queue.isEmpty()) {`

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**